src/Raven.Server/Web/RequestHandler.cs

```diff
@@ -805,5 +810,56 @@ public void AddStringToHttpContext(string str, TrafficWatchChangeType type)
         {
             HttpContext.Items["TrafficWatch"] = (str, type);
         }
+
+        public async Task WaitForDatabaseNotificationOnCluster(string databaseName, long index,
+            Dictionary<string, string> clusterNodes, // (nodeTag, url)
+            HashSet<string> databaseNodes, // nodeTags
+            Dictionary<string, ClusterRequestExecutor> reqExecutors,
+            CancellationToken token)
+        {
+
+            foreach (var (tag, nodeUrl) in clusterNodes)
+            {
+                if (databaseNodes.Contains(tag) == false)
+                    continue;
+
+                // if node is part of db topology - wait for commit and database notification
+                try
+                {
+                    ClusterRequestExecutor requestExecutor;
+                    if (reqExecutors.TryGetValue(tag, out requestExecutor) == false)
+                    {
+                        requestExecutor = reqExecutors[tag] =
+                            ClusterRequestExecutor.CreateForSingleNode(nodeUrl, Server.Certificate.Certificate, DocumentConventions.DefaultForServer);
+                    }
+
+                    using (requestExecutor.ContextPool.AllocateOperationContext(out var context))
+                    {
+                        var cmd = new WaitForDatabaseIndexNotificationCommand(index, databaseName);
+                        await requestExecutor.ExecuteAsync(cmd, context, token: token);
+                    }
+                }
+                catch (RavenException e) when (e.InnerException is HttpRequestException)
+                {
+                }
+                catch (Exception e) when (e is InvalidOperationException or AllTopologyNodesDownException)
+                {
+                    // from request executor (ThrowFailedToContactAllNodes)
+                }
+                catch (Exception e) when (e is DatabaseDoesNotExistException)
+                {
+                    // from request executor (ExecuteAsync, you check if `HandleUnsuccessfulResponse` is false, and then DatabaseDoesNotExistException.Throw).
+                }
+                catch (Exception e) when (e is DatabaseDisabledException)
+                {
+                    // when the db is being deleted
+                }
```

Comment:
instead of catch (Exception e) when (e is InvalidOperationException or AllTopologyNodesDownException)
it should be
```
catch(InvalidOperationException )
{
..
}
catch( AllTopologyNodesDownException)
{
..
}
```
etc

Reply:
Done